### PR TITLE
Add project_id to data_project

### LINKED ISSUE #162: [Feature Request] Add project_id as alternative attribute for project data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
When you want to import a project using the data source, you have to specify the project name. In my case I don't have the project name, but I have its ID. It would be useful to be able supply the project_id and let it compute all the other variables (one of which being the name) for me.

### New or Affected Resource(s)
* `data azuredevops_project `

### Potential Terraform Configuration
```hcl
data "azuredevops_project" "p_name" {
  project_name = "some_project_name"
}

output "project_computed_id" {
  value = data.azuredevops_project.p_name.id
}

data "azuredevops_project" "p_id" {
  project_id = "12345"
}

output "project_computed_name" {
  value = data.azuredevops_project.p_id.project_name
}
```

### PR DESCRIPTION
**Pull request by EliiseS**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number: closes  #162

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [x] No

## Does this introduce a breaking change?

- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by EliiseS** [comment]
It might be worth removing the `project_name` field and just having the `project_id` field accept both. Didn't do that atm because of breaking changes, but it's an easy change

File: azuredevops/internal/service/core/data_project.go [hunk 1]
```diff
@@ -17,8 +19,22 @@ func DataProject() *schema.Resource {
 				Type:     v.Type,
 				Computed: true,
 			}
+		} else {
+			v.Required = false
```
Comment (xuzhang3) [inline azuredevops/internal/service/core/data_project.go]: duplicate with `v.Optional = true`

File: azuredevops/internal/service/core/data_project.go [hunk 2]
```diff
@@ -17,8 +19,22 @@ func DataProject() *schema.Resource {
...
+			v.Optional = true
+			v.ConflictsWith = []string{
+				"project_id",
```
Comment (xuzhang3) [inline azuredevops/internal/service/core/data_project.go]: `GetProject` API doesn't care what's the param we send,  `projectName` or `projectID`,  a project will return if exist. So I think users should configure at least one of them or both.  We have two optional here:
1. Remove the `ConflictsWith`, users can configure both of them
2. Add `ConflictsWith` for `project_name` too, only one is needed here

Personal prefer optional 2

File: azuredevops/internal/service/core/data_project.go (hunk 1 above)
Comment (EliiseS) [reply azuredevops/internal/service/core/data_project.go]: But it's set to be required true, in the original definition

Comment (xuzhang3) [reply azuredevops/internal/service/core/data_project.go]: I mean `Optional = true` is equal to `Required = false` here

Comment (EliiseS) [reply azuredevops/internal/service/core/data_project.go]: Removing required results in:
image

File: azuredevops/internal/service/core/data_project.go (hunk 2 above)
Comment (EliiseS) [reply azuredevops/internal/service/core/data_project.go]: if you have conflicts on one, you already can't have both of them, so the second conflict would be redundant. I've added a test to show this :)

File: azuredevops/internal/service/core/data_project.go (hunk 1 above)
Comment (xuzhang3) [reply azuredevops/internal/service/core/data_project.go]: You get the `project_name` base schema  from `ResourceProject()`,  the origin schema is:
```
"project_name": {
  Type:             schema.TypeString,
  Required:         true,
  ValidateFunc:     validation.StringIsNotWhiteSpace,
  DiffSuppressFunc: suppress.CaseDifference,
},
```
after you change the schema at code [line 17~21](https://github.com/xuzhang3/terraform-provider-azuredevops/pull/163/files/106b8fa8bf926a2ad109ea89fd81cbba923c62f5#diff-443557ef19497038b4172b9ac825bf0dR17-R21)
```
if k == "project_name" {
  v.Required = false
  v.Optional = true
  baseSchema.Schema[k] = v
}
```
the schema will looks like :
```
"project_name": {
  Type:             schema.TypeString,
  Required:         false,
  Optional:          true,
  ValidateFunc:     validation.StringIsNotWhiteSpace,
  DiffSuppressFunc: suppress.CaseDifference,
},
```
image

Suggest rewrite the schema not extend it.

Comment (EliiseS) [reply azuredevops/internal/service/core/data_project.go]: Super valid point :) Thanks.
What do you think about my changes?

Comment (xuzhang3) [reply azuredevops/internal/service/core/data_project.go]: `DiffSuppressFunc: suppress.CaseDifference,` can  be removed.

**Review by xuzhang3 — APPROVED** [review]

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**